File: rb/features.rb
# MonsterHunterWorld SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module MonsterHunterWorldFeatures
  def self.make_feature(name)
    case name
    when "base"
      MonsterHunterWorldBaseFeature.new
    when "test"
      MonsterHunterWorldTestFeature.new
    else
      MonsterHunterWorldBaseFeature.new
    end
  end
end

File: py/features.py
# MonsterHunterWorld SDK feature factory

from feature.base_feature import MonsterHunterWorldBaseFeature
from feature.test_feature import MonsterHunterWorldTestFeature


def _make_feature(name):
    features = {
        "base": lambda: MonsterHunterWorldBaseFeature(),
        "test": lambda: MonsterHunterWorldTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
